### PR TITLE
Warn users about future change of 'extrapolate_to_dc' default value from rational to cubic

### LINKED ISSUE #522: 'extrapolate_to_dc' seems to process very slowly
Hi, I am sort of new to this, so hope this is ok to post here.

It seems like the `extrapolate_to_dc` method takes a very long time (~30 s per 5,000 point dataset) on Google colab.  Is there a particular computational step in `extrapolate_to_dc` that could be done much more efficiently/quickly?

Before finding scikit-rf, I wrote amateur code that did similar extrapolations to dc which ran much faster. 

If you are reading this, thank you for your time.  Any thoughts or help here would be greatly appreciated!!
-bob

### PR DESCRIPTION
**Pull request by jhillairet**
Fixes #522. 

Keep the default value to `rational`, but warns users this choice is now considered as deprecated and will be changed to `cubic` by default in future versions.

### CODE REVIEW

File: skrf/network.py [hunk 1]
```diff
@@ -2691,7 +2692,7 @@ def extrapolate_to_dc(self, points: int = None, dc_sparam: NumberLike = None, ki
         dc_sparam : class:`npy.ndarray` or None
             NxN S-parameters matrix at 0 Hz.
             If None S-parameters at 0 Hz are determined by linear extrapolation.
-        kind : str or int
+        kind : str or int, default is 'rational'
```
Comment (Vinc0110) [inline skrf/network.py:2695]: Maybe also write here, that this default is going to be changed to 'cubic' in the future.

File: skrf/network.py [hunk 2]
```diff
@@ -2717,6 +2718,14 @@ def extrapolate_to_dc(self, points: int = None, dc_sparam: NumberLike = None, ki
         impulse_response
         step_response
         """
+        if kind == 'warn':
```
Comment (Vinc0110) [inline skrf/network.py:2721]: Why did you introduce `kind='warn'`? Can't you just keep `kind='rational'` and check against this value to print the warning? For example:
```
if kind == 'rational':
    warnings.warn('...')
```

File: skrf/network.py [hunk 3]
```diff
@@ -2717,6 +2718,14 @@ def extrapolate_to_dc(self, points: int = None, dc_sparam: NumberLike = None, ki
...
+            warnings.warn("The default `kind` parameter will change "
+                          "from `rational` to `cubic` in future versions."
+                          "Use `kind='rational'` to keep the old default."
+                          "To silent this warning, explitly define `kind`.",
+                          category=DeprecationWarning, stacklevel=2)
```
Comment (Vinc0110) [inline skrf/network.py:2726]: Did you figure out what `stacklevel=2` does? I know, [the Python doc](https://docs.python.org/3/library/warnings.html#warnings.warn) says *"The stacklevel argument can be used by wrapper functions written in Python"*, but it that important?

**Review by Vinc0110 — CHANGES_REQUESTED** [review]
Looks good, but I made a few suggestions above.

File: skrf/network.py (hunk 2 above)
Comment (FranzForstmayr) [reply skrf/network.py:2721]: The warning would be printed every time, even if you select `kind='rational'` excplicit.

Comment (Vinc0110) [reply skrf/network.py:2721]: Oh, you're right.

File: skrf/network.py (hunk 3 above)
Comment (jhillairet) [reply skrf/network.py:2726]: I found out that `stacklevel=1` (default) didn't display any message in ipython... so I've used 2.

File: skrf/network.py (hunk 1 above)
Comment (jhillairet) [reply skrf/network.py:2695]: done

**Pull request merged**